service/insolvency_service.go

```diff
@@ -102,6 +102,18 @@ func ValidateInsolvencyDetails(svc dao.Service, transactionID string) (bool, *[]
 		return false, &validationErrors
 	}
 
+	// Check if attachment_type is resolution, if true then date_of_resolution must be present
+	hasDateOfResolution := true
+	if len(insolvencyResource.Data.Resolution.DateOfResolution) == 0 {
+		hasDateOfResolution = false
+	}
+	if hasResolutionAttachment && !hasDateOfResolution {
```

Comment:
Could we combine the checks above and just have something like:

```suggestion
	if hasResolutionAttachment && !len(insolvencyResource.Data.Resolution.DateOfResolution) == 0 {
```